app/views/users/show.html.erb

```diff
@@ -2,39 +2,42 @@
   <h1> <%= current_user.first_name %>'s Dashboard </h1>
 
   <%= button_to 'Log Out', logout_path, method: 'delete', class: "btn btn-primary mb1 bg-teal" %>
+
+  <% if current_user[:username].nil? %>
+    <%= link_to 'Connect to Github', "https://github.com/login/oauth/authorize?client_id=29dde3e3c49c550f2ade&scope=repo user",class: "badge badge-pill badge-primary" %>
+  <% end %>
```

Comment:
checks if the user is logged in. Reasoning is that the user is assigned a username only if they are logged in via github